### PR TITLE
fix(medusa): Upserting tax rates

### LINKED ISSUE #4186: Error when adding a new tax rate
### Describe the bug

Got unknown error when trying to create a new tax rate for a region

### System information

Medusa version (including plugins): ^1.11.0
@medusajs/admin: ^5.0.0
@medusajs/medusa-cli: ^1.3.14
Node.js version: 18.12.1
Database: postgresql
Operating system: windows 11
Browser (if relevant): Google Chrome

### Steps to reproduce the behavior
Go to settings -> tax settings -> choose a region -> click new tax rate -> fill name, tax rate and tax code -> create

### Expected behavior
A new tax rate created

### Screenshots
Request Body
image

API Response
image

Error log
image

### Code snippets

If applicable, add code samples to help explain your problem

### Additional context
The root cause seems like from the request validator

### PR DESCRIPTION
**Pull request by olivermrbl**
Fixes #4186

### CODE REVIEW

File: packages/medusa/src/api/routes/admin/tax-rates/update-tax-rate.ts [hunk 1]
```diff
@@ -183,8 +182,8 @@ export class AdminPostTaxRatesTaxRateReq {
   region_id?: string
 
   @IsOptional()
-  @IsType([Number, null])
-  rate?: number | null
```
Comment (srindom) [inline packages/medusa/src/api/routes/admin/tax-rates/update-tax-rate.ts:187]: thought: the design of taxes includes support for a nullable rate - the idea behind this is that if you are integrating with a third party system you may be in a situation where the numerical rate is dynamically computed through a 3rd party request. In this scenario, you use the TaxRate to tag the tax code to a product, collection, etc., the code is then used to compute and respond with the correct rate. E.g.:

- Product - Book
- TaxRate - code: book, rate: null
- Cart - items: [book]
- Total -> makes call to 3rd party with code: book and address: "some place with hyper local tax rates" -> responds with rate: 0.0932 -> This rate is used in cart total calculation

Comment (olivermrbl) [reply packages/medusa/src/api/routes/admin/tax-rates/update-tax-rate.ts:187]: Yes, this doesn't change with this PR. You can still create and update tax rates with a null rate. The changes here correct what is an inconsistency - and recently breaking - way of supporting this. 

I've added tests to ensure, we are aligned on the intended behaviour. Can you confirm?

Comment (olivermrbl) [reply packages/medusa/src/api/routes/admin/tax-rates/update-tax-rate.ts:187]: This doesn't change with this PR. You can still create and update tax rates with a null rate. The changes here correct what is an inconsistent - and recently breaking - way of supporting a nullable payload at the API level.

I've added tests to ensure we are aligned on the intended behaviour. Can you confirm?

And I've kept the explicit `null` typing for DX purposes.

**Review by srindom — APPROVED** [review]
LGTM!

**Pull request merged**